src/fieri/app/models/publish_worker.rb

```diff
@@ -0,0 +1,58 @@
+require 'sidekiq'
+require 'net/http'
+
+class PublishWorker
+  include ::Sidekiq::Worker
+
+  def perform(cookbook_name)
+    parsed = JSON.parse(get_supermarket_response(cookbook_name))
+
+    failure = false
+    publish_feedback = ''
+
+    unless cookbook_exists?(parsed, cookbook_name)
+      failure = true
+      publish_feedback += "#{cookbook_name} not found in Supermarket\n"
```

Comment:
as I scroll, I realize the folly of my comment.  I'll be deleting it and this one but putting it here in case you're just reading the email thread.